src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/models/Transition.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.models
+
+import com.amazon.opendistroforelasticsearch.jobscheduler.spi.schedule.CronSchedule
+import com.amazon.opendistroforelasticsearch.jobscheduler.spi.schedule.ScheduleParser
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParser.Token
+import org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken
+import java.io.IOException
+
+data class Transition(
+    val stateName: String,
```

Comment:
Any reason it needs to be `stateName` instead of just `state` ?

Reply:
Actually I think we should keep it as stateName and change "state" to "state_name" in the mappings too (and parser). Transition.name can be confusing since to someone new it might seem like the transition's name. Prefer having "name" only when it's something like State.name (as opposed to State.stateName).